httpsig.go

```diff
@@ -170,6 +178,27 @@ func NewSigner(prefs []Algorithm, dAlgo DigestAlgorithm, headers []string, schem
 	return s, defaultAlgorithm, err
 }
 
+// NewSignerFromSSHSigner creates a new Signer using the specified ssh.Signer
+// At the moment only ed25519 ssh keys are supported.
+// The headers specified will be included into the HTTP signatures.
+//
+// The Digest will also be calculated on a request's body using the provided
+// digest algorithm, if "Digest" is one of the headers listed.
+//
+// The provided scheme determines which header is populated with the HTTP
+// Signature.
+func NewSignerFromSSHSigner(s ssh.Signer, dAlgo DigestAlgorithm, headers []string, scheme SignatureScheme, expiresIn int64) (Signer, Algorithm, error) {
```

Comment:
nit: Let's rename `NewSignerFromSSHSigner` to `NewSSHSigner` if we're introducing a `SSHSigner` interface (see above comment).